ArduPlane/quadplane.cpp

```diff
@@ -3934,26 +3944,38 @@ Vector2f QuadPlane::landing_desired_closing_velocity()
 */
 float QuadPlane::get_land_airspeed(void)
 {
-    if (poscontrol.get_state() == QPOS_APPROACH ||
+    const auto qstate = poscontrol.get_state();
+    if (qstate == QPOS_APPROACH ||
         plane.control_mode == &plane.mode_rtl) {
-        float land_airspeed = plane.TECS_controller.get_land_airspeed();
-        if (!is_positive(land_airspeed)) {
-            land_airspeed = plane.aparm.airspeed_cruise_cm * 0.01;
+        const float cruise_speed = plane.aparm.airspeed_cruise_cm * 0.01;
+        float approach_speed = cruise_speed;
+        float tecs_land_airspeed = plane.TECS_controller.get_land_airspeed();
+        if (is_positive(tecs_land_airspeed)) {
+            approach_speed = tecs_land_airspeed;
+        } else {
+            if (qstate == QPOS_APPROACH) {
+                // default to half way between min airspeed and cruise
+                // airspeed when on the approach
+                approach_speed = 0.5*(cruise_speed+plane.aparm.airspeed_min);
+            } else {
+                // otherwise cruise
+                approach_speed = cruise_speed;
+            }
         }
-        float cruise_airspeed = plane.aparm.airspeed_cruise_cm * 0.01;
-        float time_to_landing = plane.auto_state.wp_distance / MAX(land_airspeed, 5);
+        const float time_to_landing = plane.auto_state.wp_distance / MAX(approach_speed, 5);
```

Comment:
```suggestion
        const float time_to_landing = (plane.auto_state.wp_distance - stopping_distance()) / MAX(approach_speed, 5);
```
Maybe include stopping distance? Might get some exiting feedback effect tho.... maybe use the approach speed rather than ground speed?

```suggestion
        const float time_to_landing = (plane.auto_state.wp_distance - stopping_distance(approach_speed*approach_speed)) / MAX(approach_speed, 5);
```

Reply:
added as time_to_pos1